packages/core/src/internal/proc.js

```diff
@@ -62,11 +63,6 @@ export default function proc(env, iterator, parentContext, parentEffectId, meta,
     until the generator terminates or throws
   **/
   function next(arg, isErr) {
-    // Preventive measure. If we end up here, then there is really something wrong
-    if (!mainTask.running) {
```

Comment:
I removed this because we cannot tell whether the iterator is running or not here. When `mainTask.status` is `CANCELLED`, the iterator can be either finished or still executing the finally-block.